web/src/engine/websites/MangaKawaii.ts

```diff
@@ -0,0 +1,83 @@
+        }
+            catch(error) {
+                reject(error);
+            }
+        });
+`;
+
+@Common.MangaCSS(/^https?:\/\/www\.mangakawaii\.io\/manga\/[^/]+$/, 'div.manga-view__header h1[itemprop="name headline"]')
+@Common.PagesSinglePageJS(pagesScript, 500)
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('mangakawaii', `MangaKawaii`, 'https://www.mangakawaii.io', Tags.Language.French, Tags.Source.Aggregator, Tags.Media.Manga, Tags.Media.Manhua, Tags.Media.Manhwa);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override async FetchMangas(provider: MangaPlugin) : Promise<Manga[]> {
+        const token = await this.getToken();
+        const request = new FetchRequest(new URL('changeMangaList?type=text', this.URI).href);
+        request.headers.set('X-Requested-With', 'XMLHttpRequest');
+        request.headers.set('X-CSRF-TOKEN', token);
+        const data = await FetchCSS<HTMLAnchorElement>(request, 'ul.manga-list-text li a.alpha-link');
+        return data.map(element => {
+            const bloat = element.querySelector('span');
+            if (bloat) element.removeChild(bloat);
+            return new Manga(this, provider, element.pathname, element.text.trim());
+        });
+    }
+
+    public override async FetchChapters(manga: Manga): Promise<Chapter[]> {
+        const englishChapters = await this.getChaptersFromLanguage(manga, 'en');
+        const frenchChapters = await this.getChaptersFromLanguage(manga, 'fr');
+        return [...englishChapters, ...frenchChapters];
+    }
+
+    async getChaptersFromLanguage(manga: Manga, language: string): Promise<Chapter[]> {
+        await this.changeLanguage(language);
+        const uri = new URL(manga.Identifier, this.URI);
+        const request = new FetchRequest(uri.href);
+        const firstChapter = await FetchCSS<HTMLAnchorElement>(request, 'table.table--manga tbody td.table__chapter a');
+        if (typeof firstChapter[0] === 'undefined') {
+            return [];
+        }
+
+        const uri2 = new URL(firstChapter[0].pathname, this.URI);
+        const request2 = new FetchRequest(uri2.href);
+        const data = await FetchCSS<HTMLAnchorElement>(request2, '#dropdownMenuOffset+ul li a');
+        return data.map(element => {
+            return new Chapter(this, manga, element.pathname, `${element.textContent.trim()} [${language}]`);
+        });
+
+    }
+    async getToken() {
+        const request = new FetchRequest(this.URI.href);
+        const data = await FetchCSS <HTMLMetaElement>(request, 'meta[name="csrf-token"]');
+        return data[0].content;
+    }
+
+    async changeLanguage(language) {
+        await FetchWindowScript(new FetchRequest(this.URI.href), ''); //necessary cause /lang redirect to last viewed manga
```

Comment:
This seems redundant, I think language change should work independently from the last visited URL

Reply:
Its here on purposes. Without it, sometimes languages fails to change (and i end up receiving eng chapters after a /lang/fr request and vice versa).

There is something fishy here.

Reply:
lang is server sided in a session token